js/lazy_loading.js

```diff
@@ -0,0 +1,47 @@
+//Robin Osborne's Technique https://www.robinosborne.co.uk/2016/05/16/lazy-loading-images-dont-rely-on-javascript/
+var lazy = [];
+
+registerListener('load', setLazy);
+registerListener('load', lazyLoad);
+registerListener('scroll', lazyLoad);
+registerListener('resize', lazyLoad);
+
+function setLazy(){
+    lazy = document.getElementsByClassName('lazy');
+    console.log('Found ' + lazy.length + ' lazy images');
+} 
+function lazyLoad(){
+    for(var i=0; i<lazy.length; i++){
+        if(isInViewport(lazy[i])){
+          console.log(lazy[i]);
+    			lazy[i].style.transition="opacity 2s";
+    			lazy[i].style.opacity= 1;
+            if (lazy[i].getAttribute('data-src')){
+                lazy[i].src = lazy[i].getAttribute('data-src');
+                lazy[i].removeAttribute('data-src');
+            }
+        }
+    }
+    
+    cleanLazy();
+}
+function cleanLazy(){
+    lazy = Array.prototype.filter.call(lazy, function(l){ return l.getAttribute('data-src');});
+}
+function isInViewport(el){
+    var rect = el.getBoundingClientRect();
+    
+    return (
+        rect.bottom >= 0 && 
+        rect.right >= 0 && 
+        rect.top <= (window.innerHeight || document.documentElement.clientHeight) && 
+        rect.left <= (window.innerWidth || document.documentElement.clientWidth)
+     );
+}
+function registerListener(event, func) {
+    if (window.addEventListener) {
+        window.addEventListener(event, func);
+    } else {
+        window.attachEvent('on' + event, func);
+    }
```

Comment:
`Then let’s create a simple event wiring up a helper for cross-browser support (since we’re not using jQuery):`

Robin Osborne states it is for cross-browser functionality, which I am sure is needed since all browsers do things a little differently

Reply:
Huh ok, wondering if I need to go add that to my scroll code https://github.com/USGS-VIZLAB/water-use-15/blob/master/js/scroll_tracking.js#L11

Reply:
I don't know, have you tested it on multiple browsers and found issues?

Reply:
I have not! I should though :)